core/template.js

```diff
@@ -18,19 +18,32 @@
 	 *		var tpl = new CKEDITOR.template( '<div class="{cls}">{label}</div>' );
 	 *		alert( tpl.output( { cls: 'cke-label', label: 'foo'} ) ); // '<div class="cke-label">foo</div>'
 	 *
+	 *		// Since 4.12 it is possible to pass a callback Function that returns a template.
+	 * 		var tpl2 = new CKEDITOR.template( function( data ) {
+	 * 			return data.image ? '<img src="{image}" alt="{label}"/>' : '{label}';
+	 * 		} );
+	 *		alert( tpl2.output( { image: null, label: 'foo'} ) ); // 'foo'
+	 *		alert( tpl2.output( { image: '/some-image.jpg', label: 'foo'} ) ); // <img src="/some-image.jpg" alt="foo"/>
+	 *
+	 * @since 4.12
```

Comment:
With `@since` tag like that it will appear as if the class was **introduced** in 4.12.0, which is not the case 🙂 it needs to be properly reflected in the content.

I'll take it from here and polish any minor things that will catch my attention.